app/code/community/Hps/Transit/Model/Payment.php

```diff
@@ -162,13 +162,25 @@ private function _authorize(Varien_Object $payment, $amount, $capture)
             }
 
             $response = $builder->execute();
-
-            if ($response->responseCode !== '00' || $response->responseMessage === 'Partially Approved') {
+            
+            if ($response->responseCode !== '00') {
                 // TODO: move this
                 // $this->updateVelocity($e);
-
-                if ($response->responseCode === '10' || $response->responseMessage === 'Partially Approved') {
+                $status = self::STATUS_APPROVED;
+                if($response->responseCode === '10' || $response->responseMessage === 'Partially Approved'){
                     try { $response->void()->withDescription('POST_AUTH_USER_DECLINE')->execute(); } catch (\Exception $e) {}
+                } else {
+                    $avsCvvValidation = Mage::getStoreConfig('payment/hps_transit/avs_cvv_validation');
+                    if($avsCvvValidation === true &&
+                        (!empty($response->avsResponseCode) || !empty($response->cvnResponseCode))){
+                            $declinedAvsCodes = Mage::getStoreConfig('payment/hps_transit/avs_decline_codes');
+                            $declinedCvnCodes = Mage::getStoreConfig('payment/hps_transit/cvv_decline_codes');
```

Comment:
Should be:
```
$declinedAvsCodes = explode(',', Mage::getStoreConfig('payment/hps_transit/avs_decline_codes'));
$declinedCvnCodes = explode(',', Mage::getStoreConfig('payment/hps_transit/cvv_decline_codes'));
```

Reply:
thanks.. I updated the code with explode function